cpp/src/runtime/context.cpp

```diff
@@ -50,6 +46,20 @@ jit::program_cache& context::program_cache()
 
 bool context::dump_codegen() const { return _dump_codegen; }
 
+void context::initialize_components(init_flags flags)
+{
+  // Only initialize components that haven't been initialized yet
+  auto const new_flags = flags & ~_initialized_flags;
+
+  if (has_flag(new_flags, init_flags::INIT_JIT_CACHE) && _program_cache == nullptr) {
+    _program_cache = std::make_unique<jit::program_cache>();
+  }
```

Comment:
Would `_program_cache != nullptr` before initialization?

Reply:
`_program_cache == nullptr` does not need to be checked here. removing the condition.